execute/plugin_functions.go

```diff
@@ -382,6 +382,30 @@ func mergeNonceObservations(
 	return results
 }
 
+// mergeCostlyMessages merges all costly message observations. A message is considered costly if it is observed by more
+// than `fChainDest` observers.
+func mergeCostlyMessages(
+	aos []plugincommon.AttributedObservation[exectypes.Observation],
+	fChainDest int,
+) []cciptypes.Bytes32 {
+	costlyMessages := mapset.NewSet[cciptypes.Bytes32]()
+	counts := make(map[cciptypes.Bytes32]int)
+	for _, ao := range aos {
+		for _, costlyMessage := range ao.Observation.CostlyMessages {
+			counts[costlyMessage]++
+			if counts[costlyMessage] > fChainDest {
```

Comment:
small nit: can use `if counts[costlyMessage] >= int(consensus.FPlus1(fChainDest))` here to make this condition clearer.

The goal behind this helper is to be able to easily audit where we require at least one honest observation vs. where we require an honest majority.

Reply:
Done!